### PR TITLE
docs: add about QueryBuilder::where() change in v4.3.0

### PR DESCRIPTION
**Pull request by kenjis**
**Description**
Add a breaking change reported in #7117

**Checklist:**
- [x] Securely signed commits
- [] Component(s) with PHPDoc blocks, only if necessary or adds value
- [] Unit testing, with >80% coverage
- [x] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

**Comment by tangix** [comment]
Thanks!

**Comment by kenjis** [comment]
I removed the explanation in the Query Builder page.
Because I think the sample code is not good.

It had `=`, so it can be set as `where($key, $value)`.
And I think passing a custom string to `where()` is not good practice,
and should be avoided as possible.

And I prefer `RawSql()` instead of `$escape = false`.
Because `$escape = false` might break the string.
`RawSql()` is mutable and never be changed.
`RawSql()` is easier to find out with IDE.

In conclusion, I do not want to recommend the use of custom string (and escape false).

**Comment by sclubricants** [comment]
I kind of see this explanation as unnecessary because there are uses for second parameter = ''

```php
$where = "REPLACE(column, 'somestring', '')";
$builder->where($where, '', false); 
```

Gives:

```SQL
WHERE REPLACE(column, 'somestring', '') = ''
```

At least this is how it should have been working even if it wasn't.  The documentation has always been clear that the default value of the second parameter is null. Naturally a '' should give a value of an empty string.

**Comment by kenjis** [comment]
v4.2.12:
```php
$where = "REPLACE(column, 'somestring', '')";
$builder->where($where, '', false); 
// 'SELECT * FROM "jobs" WHERE REPLACE(column, 'somestring', '') = '
```
The generated SQL string is invalid. In this case ```` does not work in v4.2.12.

**Comment by kenjis** [comment]
@sclubricants 
```php
$where = "REPLACE(column, 'somestring', '')";
$builder->where($where, '', false); 
// 'SELECT * FROM "jobs" WHERE REPLACE(column, 'somestring', '') = '
```
The code does not work in 4.3 and 4.2 (probably and before). 
Because **the value `''` will be just empty string when escape is false**.

**Comment by kenjis** [comment]
@sclubricants I think if we use `$escape = false`, the code should be:
```php
        $where = "REPLACE(column, 'somestring', '')";
        $builder->where($where, "''", false);
````

**Comment by sclubricants** [comment]
Sorry it should be escape = true

```php
        $where = "REPLACE(column, 'somestring', '')";
        $builder->where($where, "''", true);
```

If false and $value = '' then we should probably throw an error.

**Comment by kenjis** [comment]
> If $escape = false and $value = '' then we should probably throw an error.

It may be. But I don't think it is worth doing it.
Misused code that depends on the behavior where the value `''` disappears when `$escape` is false will not work.

At least the issue with BETWEEN will be *fixed* by #7155

**Review by MGatner — APPROVED** [review]

**Comment by kenjis** [comment]
The sample code in this PR is wrong.
It does work in v4.3.

The code that does not work in v4.3 needs `BETWEEN` or `EXISTS` in where() string like #7117.
But it will be *fixed* by #7155
So I close this.
